packages/effect/test/Schema/Schema/TemplateLiteralParser.test.ts

```diff
@@ -20,6 +20,18 @@ schema (SymbolKeyword): symbol`)
     expect(schema.params).toStrictEqual(params)
   })
 
+  describe("string literal based schemas", () => {
+    it("decoding", async () => {
+      const schema = Schema.TemplateLiteralParser(Schema.Literal("foo"), "bar")
+      await Util.expectDecodeUnknownSuccess(schema, "foobar", ["foo", "bar"])
```

Comment:
While this test doesn't fail with `Cannot read properties of undefined` anymore, as it does in the parent commit, it still fails on validation.

Reply:
That's expected, a successful test would be:

```ts
await Util.expectDecodeUnknownSuccess(schema, "foobar", ["foobar"])
```

because the two literals are combined into `"foobar"`.

This is another bug, namely a type-level bug:

```ts
import { Schema } from "effect"

const schema = Schema.TemplateLiteralParser(Schema.Literal("foo"), "bar")

// type Type = readonly ["foo", "bar"] // <= this is wrong
type Type = typeof schema.Type
```

Reply:
I don't think it's possible at the moment. At first glance, we would need to change how `TemplateLiteralSpan` is modeled. Specifically, we would need to modify this part: https://github.com/Effect-TS/effect/blob/ec6cb5daa843f76b4f9119b58c6a2074d7145138/packages/effect/src/SchemaAST.ts#L1096 to add support for `AST.Literal` types other than `StringKeyword | NumberKeyword`.

Reply:
What's adding to the weirdness is that the `encode` side passes the tests just fine _while using `["foo", "bar"]`_. So if we would "fix" the type error and let it `decode` to `["foobar"]`, then its own decode type isn't assignable to the thing it needs for encoding. So we would have to also change the encoding behaviour.

I believe it simply shouldn't be combining literal strings, at least not for the Parser.

Reply:
Let me investigate this further. If we need to modify `TemplateLiteralSpan`, the situation becomes a bit more complicated, as I'll need to adjust other parts of the code accordingly.